prov/gni/src/gnix_mr_cache.c

```diff
@@ -337,6 +349,236 @@ static inline int __mr_cache_lru_dequeue(
 }
 
 /**
+ * Removes a particular registration cache entry from the lru cache.
+ *
+ * @param[in] cache  a memory registration cache
+ * @param[in] entry  a memory registration cache entry
+ *
+ * @return           FI_SUCCESS, on success
+ * @return           -FI_ENOENT, on empty LRU
+ */
+static inline int __mr_cache_lru_remove(
+		gnix_mr_cache_t       *cache,
+		gnix_mr_cache_entry_t *entry)
+{
+	/* Could do some error checking to see if in cache */
+
+	dlist_remove(&entry->lru_entry);
+
+	return FI_SUCCESS;
+}
+
+/**
+ * Remove entries that have been unmapped as indicated by the notifer
+ *
+ * @param[in] cache  a memory registration cache
+ *
+ * @return           nothing
+ */
+static bool __notifier_warned = false;
+static void
+__clear_notifier_events(gnix_mr_cache_t *cache)
+{
+	int ret;
+	gnix_mr_cache_entry_t *entry;
+	RbtIterator iter;
+	uint64_t cookie;
+
+	if (!cache->attr.lazy_deregistration) {
+		return;
+	}
+
+	ret = _gnix_notifier_get_event(cache->notifier,
+				       &cookie, sizeof(cookie));
+	while (ret > 0) {
+		if (ret == sizeof(cookie)) {
+			entry = (gnix_mr_cache_entry_t *) cookie;
+			switch (entry->state) {
+			case GNIX_CES_INUSE:
+				/* First, warn that this might be a
+				 * problem.*/
+				if (__notifier_warned == false) {
+					GNIX_WARN(FI_LOG_MR,
+						  "In use memory registration"
```

Comment:
Actually, since it's a warning to the end user, it shouldn't use internal implementation terminology.  Maybe something more direct like "Registered memory region includes unmapped regions".